src/pybamm/parameters/parameter_values.py

```diff
@@ -604,10 +604,12 @@ def process_geometry(self, geometry):
 
         def process_and_check(sym):
             new_sym = self.process_symbol(sym)
-            if not isinstance(new_sym, pybamm.Scalar):
-                raise ValueError(
-                    "Geometry parameters must be Scalars after parameter processing"
-                )
+            # if not isinstance(new_sym, pybamm.Scalar) and not isinstance(
```

Comment:
can we keep this check to avoid other symbols in the mesh?

Reply:
we will have to do a search of the full tree which is fine, but we can't just take parameters because often the parameter will be something like `L_n + L_s + L_p`

Reply:
added a leaf search for this